python/cuml/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -1024,18 +1144,6 @@ class NearestNeighbors(Base,
     def effective_metric_params_(self):
         return self.metric_params or {}
 
-    def __del__(self):
```

Comment:
`__del__` on an estimator itself is a red flag. It's not universally bad, but every case I've seen of it in cuml currently is doing memory management incorrectly and has bugs. If an estimator contains non-trivial state (e.g. things that aren't wrapped in python array classes) that needs to be freed, that state should typically be wrapped in a private class and managed with `__dealloc__`/`__cinit__` or other native methods. Casting pointers to python ints to pass around and hope everything works out is a recipe for bugs.

I'll open followup issues for the other cases where this is still done in cuml.